### DISCUSSION TITLE
Method Chaining

### DISCUSSION THREAD

**Discussion by dylanmichaelryan** [livewire/livewire#995] [4. General]
Hey guys,

i don't know if this is possible, or even good thought.

Sometimes i have a few method calls at the end of my function.

For example:

```
$this->emitSelf('created-user');
$this->reset();
```

In this example it's only 2, but there are cases where i have more.

I think it would be cool if you chain them like this:

```
$this->emitSelf('created-user');
     ->reset();
```

**Comment by mblarsen** [comment]
I agree, some times that does make the code more readable.

**Comment by galler** [comment]
Agree. Would be really helpful.

**Comment by calebporzio** [comment]
Nice! Yeah, I think I agree as well.

What `$this->` properties return something non-fluent?

One's off the top of my head that don't: `emit, fill, reset, dispatchBrowserEvent, validate`

**Comment by dylanmichaelryan** [comment]
`$this->resetErrorBag();`
`$this->resetValidation();`

would also be nice

**Comment by program-mia** [comment]
This looks sooo useful in a lot of situations, I love itttt
